packages/block-library/src/embed/wp-embed-preview.native.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * WordPress dependencies
+ */
+import { memo, useMemo } from '@wordpress/element';
+import { SandBox } from '@wordpress/components';
+
+/**
+ * Checks for WordPress embed events signaling the height change when iframe
+ * content loads or iframe's window is resized.  The event is sent from
+ * WordPress core via the window.postMessage API.
+ *
+ * References:
+ * window.postMessage: https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage
+ * WordPress core embed-template on load: https://github.com/WordPress/WordPress/blob/HEAD/wp-includes/js/wp-embed-template.js#L143
+ * WordPress core embed-template on resize: https://github.com/WordPress/WordPress/blob/HEAD/wp-includes/js/wp-embed-template.js#L187
+ */
+const observeAndResizeJS = `
+     ( function() {
+         if ( ! document.body || ! window.parent ) {
+             return;
+         }
+ 
+         function sendResize( { data: { secret, message, value } = {} } ) {
+             if (
+                 [ secret, message, value ].some(
+                     ( attribute ) => ! attribute
+                 ) ||
+                 message !== 'height'
+             ) {
+                 return;
+             }
+ 
+             document
+                 .querySelectorAll( \`iframe[data-secret="\${ secret }"\` )
+                 .forEach( ( iframe ) => {
+                     if ( +iframe.height !== value ) {
+                         iframe.height = value;
+                     }
+                 } );
+ 
+             // The function postMessage is exposed by the react-native-webview library 
+             // to communicate between React Native and the WebView, in this case, 
+             // we use it for notifying resize changes.
+             window.ReactNativeWebView.postMessage(JSON.stringify( {
+                 action: 'resize',
+                 height: value,
+             }));
+         }
+ 
+         window.addEventListener( 'message', sendResize );
+ } )();`;
+
+function WpEmbedPreview( { html, ...rest } ) {
+	const wpEmbedHtml = useMemo( () => {
+		const doc = new window.DOMParser().parseFromString( html, 'text/html' );
+		const iframe = doc.querySelector( 'iframe' );
+
+		if ( iframe ) {
+			iframe.removeAttribute( 'style' );
+		}
+
+		const blockQuote = doc.querySelector( 'blockquote' );
+
+		if ( blockQuote ) {
+			blockQuote.innerHTML = '';
+		}
+
+		return doc.body.innerHTML;
+	}, [ html ] );
```

Comment:
This parsing process is similar to the web version:
https://github.com/WordPress/gutenberg/blob/b2b35ba7b89b7d62dffec9db4cbd34edb6533414/packages/block-library/src/embed/wp-embed-preview.js#L72-L87

The main difference is how the blockquote element is disabled, in the web version this is done by setting the `display` style to `none`. However, this can't be done in the native version because the library we use ([`jsdom-jscore-rn`](https://github.com/iamcco/jsdom-jscore-rn)) for parsing the DOM doesn't provide the `style` attribute. Instead the element is disabled by removing the HTML content. 

https://github.com/WordPress/gutenberg/blob/b2b35ba7b89b7d62dffec9db4cbd34edb6533414/packages/react-native-editor/src/jsdom-patches.js#L18
https://github.com/WordPress/gutenberg/blob/b2b35ba7b89b7d62dffec9db4cbd34edb6533414/packages/react-native-editor/src/jsdom-patches.js#L251-L258

blockQuote.style.display = 'none';

Reply:
I'm not sure what's the best way to test this behaviour, but what I did was creating a post with a Quote block, publishing it and embedding that WP post. I observed that it looked the same as it did on the web editor:

<img width="300" src="https://user-images.githubusercontent.com/1845482/131880726-4d0f9ba8-28e9-4710-b285-976497d9cdc3.png">

Reply:
Here what we need to test is basically that the preview is rendered the same as it's in the web version and the post/page preview. The reason for having to modify the HTML of the preview is explained thoroughly in [this issue](https://github.com/WordPress/gutenberg/issues/25344).



https://github.com/WordPress/gutenberg/issues/25344